### ISSUE TITLE
Develop Accessibility Roadmap for Lesson Contributions

### ISSUE THREAD

**Issue by kariljordan** [carpentries/strategic-plan#31]
[Goal 2](https://github.com/carpentries/strategic-plan/blob/master/STRATEGIC-PLAN.md#goal-2-intentionally-incorporate-equity-inclusion-and-accessibility-to-support-a-diverse-community) of the Carpentries Strategic Plan is to _**Intentionally incorporate equity, inclusion, and accessibility to support a diverse community.**_ One of the objectives under this goal is:
- Create a roadmap for improving the accessibility of our lessons, lesson infrastructure, online resources, and sign-up workflows for teaching workshops.

This issue tracks progress to improve the accessibility of our lessons by:

- using best practices in web accessibility in our lesson template
- providing resources for Maintainers and contributors so they can ensure the accessibility of the lessons

**Comment by fmichonneau** [comment]
The work that Zhian is doing on the lesson template will greatly simplify the tooling and mechanics for proposing changes to the lesson, in turn making them more accessible.

We have worked during Q1 2021 with a UX/UI Designer to revamp our lesson template and ensure its accessibility. 

Once the development of the next iteration of the template will have stabilized we will be able to develop resources for the Maintainers and contributors to ensure the accessibility of the lessons.

**Comment by fmichonneau** [comment]
The work on the accessibility of our lessons and lesson contributions is continuing. Zhian is improving the user experience for lesson authors and contributors with the suite of packages that builds the foundation of our lesson infrastructure ([see his talk at useR!2021](https://teamup.com/event/show/id/Y83KVafYovjbMrMJvq2o5HVwyJFNrq)).

We are also currently looking for a web developer and accessibility specialist that will ensure that the user interface for the next iteration of the lesson template is accessible. Once this is in place, we will re-evaluate the contribution process to improve its accessibility.

**Comment by fmichonneau** [comment]
The web developer will deliver the next version of the lesson template at the end of November 2021. It should then take a few weeks for Zhian to include these changes into the infrastructure he has been building. We should then be able to launch the beta version of the lesson template by the end of 2021.

The web development company building the lesson template is a leader in web accessibility so the content of the lessons should be a lot more accessible than they are now. This, in turn, should help with increasing the accessibility of our lesson contribution process.

In early 2022, we can re-evaluate and test other mechanisms that would allow our community to contribute to our lessons.

**Comment by fmichonneau** [comment]
The accessibility audit of the new lesson template is almost complete and Zhian will be able to start incorporating the new template in the lesson infrastructure tooling in early 2022. During the first quarters of 2022, we will work on beta testing the new lesson template on a subset of our lessons and create a timeline for switching all our lessons to the new infrastructure.

Additionally, we have received funding (to be officially announced) that will allow us to improve the accessibility of many of our processes and workflows including lesson contributions during 2022 and 2023.

**Comment by zkamvar** [comment]
We have incorporated accessible HTML, CSS, and JavaScript that was contracted in 2021 into the new lesson infrastructure in January 2022 (https://carpentries.org/blog/2022/01/live-lesson-infrastructure/). Pre-flight accessibility checks for links (descriptive text and alternative text for images) have been added to the workbench.

We are still exploring methods for automated testing of these components as the frontend continues to evolve.

**Comment by anenadic** [comment]
The work on https://github.com/carpentries/lesson-development-training also falls under this I believe - not just contributions to existing lessons but also creating new lessons and making that accessible. Loads of great work already done on this topic by @tobyhodges.

**Comment by zkamvar** [comment]
No update for 2022-Q4 other than the Beta Phase for The Workbench is set to begin on 2022-12-12 and the lesson infrastructure is set to transition in 2023-Q2, which will address this issue.

**Comment by ErinBecker** [comment]
In May 2023, all 43 official Carpentries lessons were transitioned to The Workbench. 🎉 Read more about The Workbench release and it's impact on Maintainers and contributors here: https://carpentries.org/blog/2023/08/celebrating-carpentries-workbench/
